cmd/lifecycle/restorer.go

```diff
@@ -126,12 +160,32 @@ func (r restoreArgs) restore(layerMetadata platform.LayersMetadata, group buildp
 		LayersMetadata:        layerMetadata,
 	}
 
-	if err := restorer.Restore(cacheStore); err != nil {
+	if err := restorer.Restore(cacheStore, previousImage); err != nil {
```

Comment:
What was the reason we didn't sort of follow the `LayerMetadataRestorer` pattern with `SBOMFileRestorer`?

Reply:
**My humble opinion:**
I didn't think the complexity introduced in `restorer.go` was worth the tradeoff of hiding behind another abstraction. As a newcomer to the code, it took me quite a bit of time to understand what the `LayerMetadataRestorer` was doing exactly, and how it related to the rest of the code. I was trying to avoid the same with this BOM restoration code.

But the biggest reason was the performance implications. The BOM restoration code makes 2 asynchronous calls to restore layers, which happens at the same time that buildpack layers are being restored.

```go
				g.Go(func() error {
					return r.restoreCacheLayer(cache, cachedLayer.SHA)
				})
```

If we moved these calls to separate abstractions, it would be hard/clumsy to keep sure that they're having in parallel, for future developers